src/common.rs

```diff
@@ -149,6 +145,21 @@ pub fn solve_quadratic(c0: f64, c1: f64, c2: f64) -> ArrayVec<f64, 2> {
     result
 }
 
+/// Find real roots of linear equation.
+///
+/// Return values of x for which c0 + c1 x = 0.
+pub fn solve_linear(c0: f64, c1: f64) -> ArrayVec<f64, 1> {
```

Comment:
Personally, I like the idea of keeping it consistent. You're absolutely right that what I have is the equivalent of option though. I think it comes down to personal preference, but since you maintain this repo, I'll let you have the final say.